opus/application/static_media/js/cart.js

```diff
@@ -510,38 +511,62 @@ var o_cart = {
             o_browse.showPageLoaderSpinner();
 
             // redux: and nix this big thing:
-            $.ajax({ url: "/opus/__cart/view.html",
-                success: function(html) {
-                    // this div lives in the in the nav menu template
-                    $("#op-download-options-container", "#cart").hide().html(html).fadeIn();
-
-                    // Init perfect scrollbar when .op-download-options-product-types is rendered.
-                    o_cart.downloadOptionsScrollbar = new PerfectScrollbar(".op-product-type-table-body", {
-                        minScrollbarLength: opus.minimumPSLength,
+            let hash = o_hash.getHash();
+
+            // Figure out which product types are not selected
+            let notSelectedProductsList = $("#op-cart-summary .op-download-options-product-types :checkbox:not(:checked)");
+            let notSelectedProductInfoSlugName = [];
+            $.each(notSelectedProductsList, function(index, linkObj) {
+                notSelectedProductInfoSlugName.push($(linkObj).val());
+            });
+            let notSelected = "";
+            if (hash !== "") {
+                notSelected = "&";
+            }
+            notSelected += "unselected_types=" + notSelectedProductInfoSlugName.join();
+            let selected = o_cart.getDownloadFiltersChecked();
+            // make sure that if there are not types because this was a refresh, we reset the types to 'all'
```

Comment:
I'm not understanding this...how does "types=" differ between a refresh and just having no types selected?

Reply:
because types= means that the types are all unselected; having no types in the get.request will default to 'all' 
line 101 in views.py:     product_types_str = request.GET.get('types', 'all')